test/tests/customhighlights.js

```diff
@@ -96,4 +96,56 @@ describe("Custom highlights", function () {
 		client.compileCustomHighlights();
 		expect(client.highlightRegex).to.be.null;
 	});
+
+	// tests for highlight exceptions
+	client.config.clientSettings.highlights = "foo, @all,   sp ace   , 고";
+	client.config.clientSettings.highlightExceptions = "foo bar, bar @all, test sp ace test";
+
+	it("should NOT highlight due to highlight exceptions", function () {
```

Comment:
this `NOT` should be in next test i think?

Reply:
No, it's not the regex where the negation happens but the code, so the regex needs to match things that actually should not highlight and vice versa.
Or in other words, if this test fails, the user would get a highlight because the exception code fails.